src/core/Link/Link.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import { useStyles } from './styles';
+
+interface LinkProps {
```

Comment:
can we extend the Props of NavLink...that way we don't have to have target and pathname separately
```suggestion
interface LinkProps extend LinkBaseProps{
```
You have to take NavLinkProps from base.ts where you'll import it like
```
import { NavLinkProps } from 'react-router-dom'

export type LinkBaseProps = NavLinkProps;

```